app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php

```diff
@@ -135,12 +135,13 @@ private function getAdditionalFields(array $objectArray): array
      * Check whether price has percentage value.
      *
      * @param array $priceRow
-     * @return int|null
+     *
+     * @return float|null
      */
-    private function getPercentage(array $priceRow): ?int
+    private function getPercentage(array $priceRow): ?float
```

Comment:
As mentioned in previous incarnation of this PR - type hinting needs to be removed so that any of int/float is preserved. Type casting within function should be done with `+ 0` or `* 1`.

Also, please squash all changes into a single commit.

Reply:
@orlangur , okay, but is there a real reason to avoid float type hint?
As I saw, there were errors in tests because of removed isset statement

Reply:
Please review.

Reply:
> a real reason to avoid float type hint?

Just that there is no reason to convert int.